georaster-layer-for-leaflet.js

```diff
@@ -442,6 +446,56 @@ const GeoRasterLayer = L.GridLayer.extend({
     }
   },
 
+  /**
+   * The callback used to determine the colour based on the values of each pixel
+   *
+   * @callback pixelValuesToColor
+   * @param {array | number} values - The pixel value
+   * @returns {string} - Any valid CSS color string
+   */
+
+  /**
+   * Redraws the active map tiles updating the pixel values using the supplied callback
+   * @param {pixelValuesToColor} pixelValuesToColor - Callback that handles getting the pixel color
+   */
+  updateColors(pixelValuesToColor) {
+    if (!pixelValuesToColor) {
+      console.error("Missing pixelValuesToColor function");
+      return this;
+    }
+
+    if (this.debugLevel >= 1) console.log("Start updating active tile pixel values");
+
+    // update option to ensure correct colours at other zoom levels.
+    this.options.pixelValuesToColorFn = pixelValuesToColor;
+
+    const tiles = this.getActiveTiles();
+    if (!tiles) {
+      console.error("No active tiles available");
+      return this;
+    }
+
+    if (this.debugLevel >= 1) console.log("Active tiles fetched", tiles);
+
+    tiles.forEach(tile => {
+      const { coords, el } = tile;
+      this.drawTile(tile, coords, el.getContext("2d"));
+    });
+    if (this.debugLevel >= 1) console.log("Finished updating active tile colours");
+    return this;
+  },
+
+  getTiles() {
+    return this._tiles;
+  },
+
+  getActiveTiles() {
+    const tiles = this.getTiles();
+
+    // only return valid tiles
+    return filter(tiles, tile => this._isValidTile(tile.coords));
```

Comment:
could we change to ?
```js
return tiles.filter(tile => this._isValidTile(tile.coords));
```

Reply:
Also, just double-checking.  Does this.getTiles() always return an Array?

Reply:
So I used `_.filter` because `this.getTiles()` returned an object and it works on any collection.  I'll do some digging and see what the possibilites are for a pure JS option.

Reply:
So this the `getTiles()` method returns a collection object of tiles.  We could turn the object colleciton into an array using `Object.values(tiles)` in the `getTiles()` method. Or do the same thin in the `getActiveTiles()` method?

My feeling is doing it at the source is better as I expect to mostly need an array rather than an object.

Reply:
Sounds great!